### PR TITLE
[FIX] product_sold_by_delivery_week: work with lower permissions

### PR DESCRIPTION
**Pull request by yajo**
A user with basic stock and sale permissions couldn't validate a picking because it meant writing a field into the product. Now that write is issued with `sudo()`.

I refactored a bit the tests. They were testing very weird scenarios that would never happen in the real world. And, what's more interesting, they were not testing the real world scenarios.

I lowered the permissions in tests, to make sure that the module works with the basic permissions.

Finally, I also changed some spelling mistakes that I simply could not help but change.

@moduon MT-5114

### CODE REVIEW

File: product_sold_by_delivery_week/README.rst [hunk 1]
```diff
@@ -57,7 +57,7 @@ info in other views like product recommendations, this is not needed.)
 Usage
 =====
 
-Once configured, hints will apperar as an optional column in sale order
+Once configured, hints will aperar as an optional column in sale order
```
Comment (chienandalu) [inline product_sold_by_delivery_week/README.rst]: :thinking: I see I'm not the only typo blind in the house :smile: 
```suggestion
Once configured, hints will appear as an optional column in sale order
```

Or maybe we could just use *show up*

**Review by chienandalu — COMMENTED** [review]
Thanks @yajo :wink: 

Tiny remark:

File: product_sold_by_delivery_week/README.rst (hunk 1 above)
Comment (yajo) [reply product_sold_by_delivery_week/README.rst]: haah ooops

**Review by Shide — APPROVED** [review]
Code review LGTM

**Review by chienandalu — APPROVED** [review]

**Review by EmilioPascual — APPROVED** [review]

**Comment by yajo** [comment]
Thanks everyone.

/ocabot merge patch

**Pull request merged**